RecoTracker/MkFitCMS/src/MkStdSeqs.cc

```diff
@@ -586,39 +594,131 @@ namespace mkfit {
+      int npixhits = t.nPixelDecoded(enhits);
+      int enlyrs = t.nLayersByTypeEncoded(trk_inf);
+      int npixlyrs = t.nPixelDecoded(enlyrs);
+      int nmatlyrs = t.nTotMatchDecoded(enlyrs);
+      int llyr = t.getLastFoundHitLyr();
+      int lplyr = t.getLastFoundPixelHitLyr();
+      float invpt = t.invpT();
+      float invptmin = 1.43;  // min 1/pT (=1/0.7) for full filter on (npixhits<=3 .or. npixlyrs<=3)
+      float d0BS = t.d0BeamSpot(bspot.x, bspot.y);
+      float d0_max = 0.1;  // 1 mm
+
+      bool endsInsidePix = (llyr == 2 || llyr == 18 || llyr == 45);
+      bool lastInsidePix = ((0 <= lplyr && lplyr < 3) || (18 <= lplyr && lplyr < 20) || (45 <= lplyr && lplyr < 47));
+      return !(((npixhits <= 3 || npixlyrs <= 3) && endsInsidePix &&
+                (invpt < invptmin || (invpt >= invptmin && std::abs(d0BS) > d0_max))) ||
+               ((npixlyrs <= 3 && nmatlyrs <= 6) && lastInsidePix && llyr != lplyr && std::abs(d0BS) > d0_max));
     }
 
+    /// quality filter tuned for pixelLess iteration during forward search
+    template <class TRACK>
+    bool qfilter_pixelLessFwd(const TRACK &t, const MkJob &j) {
+      const BeamSpot &bspot = j.m_beam_spot;
+      const TrackerInfo &tk_info = j.m_trk_info;
+      float d0BS = t.d0BeamSpot(bspot.x, bspot.y);
+      float d0_max = 0.05;  // 0.5 mm
+
+      int encoded;
+      encoded = t.nLayersByTypeEncoded(tk_info);
+      int nLyrs = t.nTotMatchDecoded(encoded);
+      encoded = t.nHitsByTypeEncoded(tk_info);
+      int nHits = t.nTotMatchDecoded(encoded);
+
+      int seedReduction = (t.getNSeedHits() <= 5) ? 2 : 3;
+
+      float invpt = t.invpT();
+      float invptmin = 1.11;  // =1/0.9
+
+      float thetasym = std::abs(t.theta() - Const::PIOver2);
+      float thetasymmin = 1.11;  // -> |eta|=1.45
+
+      return (((t.nFoundHits() - seedReduction >= 4 && invpt < invptmin) ||
+               (t.nFoundHits() - seedReduction >= 3 && invpt > invptmin && thetasym <= thetasymmin) ||
+               (t.nFoundHits() - seedReduction >= 4 && invpt > invptmin && thetasym > thetasymmin)) &&
+              !((nLyrs <= 4 || nHits <= 4) && std::abs(d0BS) > d0_max && invpt < invptmin));
+    }
+
+    /// quality filter tuned for pixelLess iteration during backward search
+    template <class TRACK>
+    bool qfilter_pixelLessBkwd(const TRACK &t, const MkJob &j) {
+      const BeamSpot &bspot = j.m_beam_spot;
+      const TrackerInfo &tk_info = j.m_trk_info;
+      float d0BS = t.d0BeamSpot(bspot.x, bspot.y);
+      float d0_max = 0.1;  // 1 mm
+
+      int encoded;
+      encoded = t.nLayersByTypeEncoded(tk_info);
+      int nLyrs = t.nTotMatchDecoded(encoded);
+      encoded = t.nHitsByTypeEncoded(tk_info);
+      int nHits = t.nTotMatchDecoded(encoded);
+
+      float invpt = t.invpT();
+      float invptmin = 1.11;  // =1/0.9
+
+      float thetasym = std::abs(t.theta() - Const::PIOver2);
+      float thetasymmin_l = 0.80;  // -> |eta|=0.9
```

Comment:
These magic numbers are particularly confusing because they aren't integers corresponding to geometric differences. They definitely should be configurable.

Reply:
same response as in `qfilter_pixelLessFwd`

Reply:
> They definitely should be configurable.

similarly, the purpose of this PR is to make the choice of selector functions configurable, as done at the PR submission.